### PR TITLE
Exporter jobs now pull from remote-config

### PR DESCRIPTION
**Pull request by pwnage101**
secure PR: https://github.com/edx-ops/analytics-secure/pull/457

### CODE REVIEW

File: dataeng/resources/org-exporter-worker.sh [hunk 1]
```diff
@@ -2,18 +2,11 @@ TODAY=$(date +%d)
 
 env | sort
 
-# Uncomment the following section only while working DENG-382
-#
-# Fetch the latest config:
-#mkdir ${WORKSPACE}/remote-config
-#aws s3 cp s3://edx-remote-config/prod/lms.yml ${WORKSPACE}/remote-config/lms.yml
-#aws s3 cp s3://edx-remote-config/prod/studio.yml ${WORKSPACE}/remote-config/studio.yml
```
Comment (pwnage101) [inline dataeng/resources/org-exporter-worker.sh:10]: Jokes on me for thinking the solution was this simple.  It was complicated enough that I ended up having to break it out to a separate file (remote-config.sh).

File: dataeng/resources/remote-config.sh [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+#
+# Fetch the latest studio and LMS configs for edx and edge.
+#
+# For each deployment:
+#
+# 1. Assume the cross-account role specifically made for this purpose.
+# 2. Fetch the encrypted configs.
+# 3. Login to Vault.
+# 4. Fetch the decryption keys from Vault.
+# 5. Decrypt the configs
```
Comment (pwnage101) [inline dataeng/resources/remote-config.sh:11]: I manually tested every step of this script on jenkins.

File: dataeng/resources/remote-config.sh [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    asym_crypto_yaml decrypt-encrypted-yaml \
+        --private_key_path ${DECRYPTION_KEY_PATH} \
+        --secrets_file_path ${WORKSPACE}/remote-config/prod-${DEPLOYMENT}/lms.encrypted.yml \
+        --outfile_path ${WORKSPACE}/remote-config/prod-${DEPLOYMENT}/lms.yml
+    asym_crypto_yaml decrypt-encrypted-yaml \
+        --private_key_path ${DECRYPTION_KEY_PATH} \
+        --secrets_file_path ${WORKSPACE}/remote-config/prod-${DEPLOYMENT}/studio.encrypted.yml \
+        --outfile_path ${WORKSPACE}/remote-config/prod-${DEPLOYMENT}/studio.yml
+
+    # For good measure, delete the decryption key.  The wsCleanup() post-build step would delete anyway it if this fails
```
Comment (brianhw) [inline dataeng/resources/remote-config.sh:95]: nit:   "anyway it" => "it anyway" ?

File: dataeng/resources/remote-config.sh [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+# 3. Login to Vault.
+# 4. Fetch the decryption keys from Vault.
+# 5. Decrypt the configs
+
+set -xe
+
+env | sort
+
+# Install dependencies early in this script.
+pip install asym-crypto-yaml
```
Comment (brianhw) [inline dataeng/resources/remote-config.sh:18]: I'm just curious about the other dependencies in this script, and where they come from.    Those include aws cli, vault, and jq.   Can we just assume these are on the Jenkins already?

**Review by brianhw — APPROVED** [review]
Looks plausible to me.   A nit is not blocking, and a question about prereqs is also not blocking.

File: dataeng/resources/remote-config.sh (hunk 4 above)
Comment (pwnage101) [reply dataeng/resources/remote-config.sh:18]: * awscli is installed on the machine
* jq: https://github.com/edx/configuration/pull/6056 (I manually installed it on the machine)
* vault... drat.  I guess I only tested the vault step on my local machine.

Comment (pwnage101) [reply dataeng/resources/remote-config.sh:18]: https://github.com/edx/configuration/pull/6076

**Comment by pwnage101** [comment]
An actual test run that, at the time of this writing, has made it all the way to edge reporting: http://jenkins.analytics.edx.org/job/test-analytics-exporter-worker/20/console

**Comment by brianhw** [comment]
LGTM.

**Pull request merged**